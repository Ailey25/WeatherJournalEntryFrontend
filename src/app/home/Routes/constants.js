export const APP_URL = {
  HOME_TAB: '/home',
  JOURNALS_TAB: '/journals',
  SETTINGS_TAB: '/user-settings',
  LOGIN: '/login',
  REGISTER: '/register',
};
