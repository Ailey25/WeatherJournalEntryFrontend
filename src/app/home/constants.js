export const BASE_URL = '/api/values';

// Login and register
// also used as input ids for login/register so " " instead of ' '
export const USERNAME = "username";
export const FIRSTNAME = "firstname";
export const LASTNAME = "lastname";
export const PASSWORD = "password";

// home
export const TOGGLE_TEMP_UNIT = 'TOGGLE_TEMP_UNIT';

// journal
// also used as input ids for journal so " " instead of ' '
export const CREATE = "create";
export const EDIT = "edit";
export const CITY_ID = 'CITY_ID';
export const CITY_NAME = 'CITY_NAME';

// submission validation
const SUCCESS = 'SUCCESS';
const EMPTY = 'EMPTY';
const NON_ALPHABET = 'NON_ALPHABET';
const INVALID_COMMA = 'INVALID_COMMA';
const NON_NUMBER = 'NON_NUMBER';

export const CITY_NAME_VALIDATION_STATUS = {
  [SUCCESS]: 1,
  [EMPTY]: 2,
  [INVALID_COMMA]: 3,
  [NON_ALPHABET]: 4,
};

export const CITY_ID_VALIDATION_STATUS = {
  [SUCCESS]: 1,
  [EMPTY]: 2,
  [NON_NUMBER]: 3,
};

export const CITY_NAME_VALIDATION_MESSAGE = {
  [CITY_NAME_VALIDATION_STATUS.EMPTY]: 'City name value cannot be empty',
  [CITY_NAME_VALIDATION_STATUS.INVALID_COMMA]: 'Enter country code after comma or remove comma',
  [CITY_NAME_VALIDATION_STATUS.NON_ALPHABET]: 'City name and/or country code can only contain alphabets',
};

export const CITY_ID_VALIDATION_MESSAGE = {
  [CITY_ID_VALIDATION_STATUS.EMPTY]: 'City id value cannot be empty',
  [CITY_ID_VALIDATION_STATUS.NON_NUMBER]: 'City ID must be numbers',
};

// weather stamp
export const CELCIUS = 'C';
export const FAHRENHEIT = 'F';
