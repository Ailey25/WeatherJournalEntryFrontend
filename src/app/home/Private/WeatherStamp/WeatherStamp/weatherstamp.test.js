import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount, render } from 'enzyme';

import { WeatherStamp } from './index';
import { CREATE, CELCIUS } from '../../../constants';

configure({ adapter: new Adapter() })

const defaultProps = () => ({
  mode: {CREATE},
  isLoading: false,
  cityName: '',
  unit: CELCIUS,
  temp: NaN,
  weather: []
});

const testProps1 = () => ({
  mode: CREATE,
  isLoading: false,
  cityName: 'test city name',
  unit: CELCIUS,
  temp: '0',
  weather: [
    {
      icon: '10d',
      weatherId: '1',
      description: 'Really cold'
    }, {
      icon: '03n',
      weatherId: '2',
      description: 'Rain everywhere'
    }
  ]
});

describe('WeatherStamp [shallow]', () => {
  it('renders without crashing', () => {
    let wrapper = shallow(<WeatherStamp />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('WeatherStamp [mount]', () => {
  it('renders props correctly', () => {
    let wrapper = mount(<WeatherStamp {...testProps1()} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('WeatherStamp with default props', () => {
  it('[shallow]', () => {
    let wrapper = shallow(<WeatherStamp {...defaultProps()} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('[mount]', () => {
    let wrapper = mount(<WeatherStamp {...defaultProps()} />);

    expect(wrapper).toMatchSnapshot();
  });
});
