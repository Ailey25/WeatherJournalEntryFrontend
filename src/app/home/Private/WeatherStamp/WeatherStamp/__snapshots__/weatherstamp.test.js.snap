// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherStamp [mount] renders props correctly 1`] = `
<WeatherStamp
  cityName="test city name"
  isLoading={false}
  mode="create"
  temp="0"
  unit="C"
  weather={
    Array [
      Object {
        "description": "Really cold",
        "weatherId": "1",
      },
      Object {
        "description": "Rain everywhere",
        "weatherId": "2",
      },
    ]
  }
/>
`;

exports[`WeatherStamp [shallow] renders without crashing 1`] = `
<div>
  <label />
  <br />
  <label>
    <sup>
      °
    </sup>
  </label>
  <br />
  <div />
</div>
`;

exports[`WeatherStamp with default props [mount] 1`] = `
<WeatherStamp
  cityName=""
  isLoading={false}
  mode={
    Object {
      "CREATE": "create",
    }
  }
  temp={NaN}
  unit="C"
  weather={Array []}
>
  <div>
    <label />
    <br />
    <label>
      NaN
      <sup>
        °
        C
      </sup>
    </label>
    <br />
    <div />
  </div>
</WeatherStamp>
`;

exports[`WeatherStamp with default props [shallow] 1`] = `
<div>
  <label />
  <br />
  <label>
    NaN
    <sup>
      °
      C
    </sup>
  </label>
  <br />
  <div />
</div>
`;
